tests/components/homekit_controller/common.py

```diff
@@ -141,4 +177,16 @@ def __init__(self, hass, entity_id, pairing, accessory):
     fire_service_discovered(hass, SERVICE_HOMEKIT, discovery_info)
     await hass.async_block_till_done()
 
-    return Helper(hass, '.'.join((domain, 'testdevice')), pairing, accessory)
+    # We can't guarantee what our entity_id will be so we have to find it
+    # in hass after its been created. (The cover entity in particular unsets
+    # its name, meaning its unique_id is used and that isn't stable enough to
```

Comment:
Can't we patch the thing that gives the unique_id so that we get a stable entity_id? We should avoid interfacing with the entity.

Reply:
I like this idea. It uses name sometimes and unique_id others but I guess I could patch both. I’d have to patch the class as I won’t have access to the instance before it’s too late. I’ve never patched the class then instance it so I’ll have to give it a try...

Reply:
Ooh and actually won’t even need the monkey patch after my original PR because that sets the name from the homekit data during __init__ (waiting till first update is a bug really)